### PR TITLE
1.0.rc2

### PR DESCRIPTION
**Pull request by adrinjalali**
Somehow the bot hasn't picked up on the tag, creating this PR for the RC.

### CODE REVIEW

**Comment by glemaitre** [comment]
FYI the issue is:

```python
    @pytest.mark.parametrize(
        "klass",
        [
            SGDClassifier,
            SparseSGDClassifier,
            SGDRegressor,
            SparseSGDRegressor,
            SGDOneClassSVM,
            SparseSGDOneClassSVM,
        ],
    )
    @pytest.mark.parametrize("fit_method", ["fit", "partial_fit"])
    @pytest.mark.parametrize(
        "params, err_msg",
        [
            ({"alpha": -0.1}, "alpha must be >= 0"),
            ({"penalty": "foobar", "l1_ratio": 0.85}, "Penalty foobar is not supported"),
            ({"loss": "foobar"}, "The loss foobar is not supported"),
            ({"l1_ratio": 1.1}, r"l1_ratio must be in \[0, 1\]"),
            ({"learning_rate": "<unknown>"}, "learning rate <unknown> is not supported"),
            ({"nu": -0.5}, r"nu must be in \(0, 1]"),
    def SGDOneClassSVM(**kwargs):
        _update_kwargs(kwargs)
>       return linear_model.SGDOneClassSVM(**kwargs)
E       TypeError: __init__() got 2 unexpected keyword arguments
```

And this is repeated for the different estimators in the parametrization. I don't really get why we don't catch it in our CI. At least for the CPython implementation. I am unsure yet but it might just due to a brittle error message match.

**Comment by glemaitre** [comment]
It was already shown in our `pypy` builds but we completely disregarded it.

**Comment by glemaitre** [comment]
And it seems that I wrote that test :). Whoopsy.

**Comment by adrinjalali** [comment]
> Non noarch packages should have python requirement without any version constraints.

How am I supposed to say I don't want python3.6?

**Comment by adrinjalali** [comment]
https://github.com/scikit-learn/scikit-learn/pull/20978 should fix the PyPy issue.

**Comment by adrinjalali** [comment]
@jakirkham do you know why this is not skipping 3.6 builds?

**Comment by glemaitre** [comment]
The tests are passing at least.

**Comment by adrinjalali** [comment]
I didn't know this could happen: 

> The job exceeded the maximum log length, and has been terminated.

The travis jobs [failing here](https://app.travis-ci.com/github/conda-forge/scikit-learn-feedstock/jobs/536434831)

**Comment by glemaitre** [comment]
We innovate at each release apparently. We need to remove the `-v` when launching `pytest`.

**Comment by jakirkham** [comment]
> do you know why this is not skipping 3.6 builds?

We need to re-render ;)

@conda-forge-admin, please re-render

**Comment by ogrisel** [comment]
@conda-forge-admin, please re-render

**Comment by ogrisel** [comment]
> We innovate at each release apparently. We need to remove the -v when launching pytest.

I pushed a commit to do so. Let's see how it goes.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,18 +1,18 @@
-{% set version = "0.24.2" %}
+{% set version = "main" %}
```
Comment (ogrisel) [inline recipe/meta.yaml:1]:
```suggestion
{% set version = "1.0.rc1" %}
```

**Comment by ogrisel** [comment]
@conda-forge-admin, please re-render

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
+{% set version = "1.0.rc1" %}
 
 package:
   name: scikit-learn
   version: {{ version }}
 
 source:
   url: https://github.com/scikit-learn/scikit-learn/archive/{{ version }}.tar.gz
-  sha256: 642fb016bfe4bb7539ba6bf4e6dd5a95d2d25638387040b0f5eefdb84a840297
+  sha256: af4fbc0a5d25f7cee8c16640b8d5e5130aad60ef563a06d234b20845b1a35e29
```
Comment (ogrisel) [inline recipe/meta.yaml:9]:
```suggestion
  sha256: 0584ba027887771b872d022f41e84ebf6724700d5eccb1d1003f18e3a368eb60
```

**Comment by adrinjalali** [comment]
This is expected to fail @ogrisel , the test fix for pypy is not on the release branch yet.

I'm not too attached to having the conda package out for the RC, are you?

**Comment by glemaitre** [comment]
> I'm not too attached to having the conda package out for the RC, are you?

I would be in this case. The idea of the RC is to catch as many regressions as possible. Having the conda package could allow being tested on CIs that are installing from `conda-forge` (hoping that they installed RCs on some builds.)

**Comment by adrinjalali** [comment]
so do you wanna merge this to have the builds for cpython, or should we do an RC2?

**Comment by adrinjalali** [comment]
Here's the PR to start an RC2: https://github.com/scikit-learn/scikit-learn/pull/21008

**Comment by adrinjalali** [comment]
We get a persistent timeout on travis for python3.7. @ogrisel would you have time to check that before the release? I'm merging this to get the binaries for the other platforms out for RC2.

Failing: CONFIG=linux_ppc64le_numpy1.19python3.7.____73_pypypython_implpypy UPLOAD_PACKAGES=True PLATFORM=linux-ppc64le DOCKER_IMAGE=quay.io/condaforge/linux-anvil-ppc64le

**Comment by jakirkham** [comment]
Sorry for missing this, but was this meant to be merged into `master` or the [`rc` branch]( https://github.com/conda-forge/scikit-learn-feedstock/tree/rc )? Asking as the latter would have put this under an [`rc` label]( https://github.com/conda-forge/scikit-learn-feedstock/blob/3680e984dbd95e142d5ea8fba965fc4d5d4923fc/recipe/conda_build_config.yaml#L1-L2 ) as opposed to `main`.

**Comment by adrinjalali** [comment]
Oh, I thought it would check the version and would work the way `pip` works. That's really bad, I'll revert this on `master` now.

**Comment by adrinjalali** [comment]
UPDATE: I'll be off till later today, please revert this on `master` if you have time, since now everybody who's installing with conda-forge is getting the RC. Otherwise I'll do later today.

**Comment by ogrisel** [comment]
I can try to do it.

**Pull request merged**